src/SkiddleBase.php

```diff
@@ -125,24 +136,116 @@ public function formatForLog($data = [], $unset = false)
                     unset($data[$k][$old]);
                 }
             }
-            foreach ($reformat_venues as $old => $new) {
+            foreach ($reformatVenues as $old => $new) {
                 $data[$k][$new] = $ticket['venue'][$old];
                 if ($unset) {
                     unset($data[$k]['venue'][$old]);
                 }
             }
             $data[$k]['Town'] = $ticket['venue']['town']; //:thinking_face:
-            foreach ($reformat_times as $old => $new) {
+            foreach ($reformatTmes as $old => $new) {
                 $data[$k][$new] = $ticket['openingtimes'][$old];
                 if ($unset) {
                     unset($data[$k]['openingtimes'][$old]);
                 }
+                $data[$k][$new] = $ticket['openingTimes'][$old];
+                if ($unset) {
+                    unset($data[$k]['openingTimes'][$old]);
+                }
             }
         }
 
         return $data;
     }
 
+    /**
+     * v3 API - helper function
+     * @param array  $responseData event data in the response
+     * @param array  $renameKeys keys to remap
+     * @return array event data with transformed keys
+     */
+    private function recursiveRenameKeys(array $array, array $renameKeys): array
+    {
+        $newArray = [];
+
+        foreach ($array as $key => $value) {
+
+            if (array_key_exists($key, $renameKeys)) {
+                $newKey = $renameKeys[$key];
+            } else {
+                $newKey = $key;
+            }
+
+            if (is_array($value)) {
+                $newArray[$newKey] = $this->recursiveRenameKeys($value, $renameKeys);
+            } else {
+                $newArray[$newKey] = $value;
+            }
+        }
+
+        return $newArray;
+    }
+
+    /**
+     * v3 API - backward compatability workaround for old requests
+     * @param array  $responseData The results to group
+     * @return array The formatted results
+     */
+    public function transformResponseKeys(array $responseData): array
```

Comment:
Is this code used outside of the library? I can't see a reference to it within the SDK.

I realise there is existing code for mapping things, but I do think this would be better in its own class within the SDK. Perhaps `recursiveRenameKeys()` can go with it? Just shooting from the hip, but how about something like this?

```php
class V3Transformer
{    
    public $keys = [
        /** list of keys */
    ];

    public function transform(array $responseData): array
    {
        return $this->renameKeys($responseData, $this->keys);
    }

    private function renameKeys(array $responseData, array $renameKeys): array
    {
        $newArray = [];

        foreach ($responseData as $key => $value) {
            $newKey = array_key_exists($key, $renameKeys) ? $renameKeys[$key] : $key;

            $newArray[$newKey] = is_array($value) ? $this->renameKeys($value, $renameKeys) : $value;
        }

        return $newArray;
    }
}
```

Ideally, we'd have `renameKeys()` and `transform()` defined in a parent `Transformer` class so we could use it elsewhere (i.e. V1Transformer), but I think the current API would need a larger redesign to accommodate this direction.

To use it, you can:

```php
$transformer = new V3Transformer();
$transformedResponse = $transformer->transform($response);
```

Reply:
I added this function near the start of these SDK updates, which could be called within individual pages if backwards compatibility was needed - but turns out i only really used it on 1 page.
So I've made changes to that 1 page which has now made the transformResponseKeys function redundant.... so i've removed it